source/fundamentals/typescript.txt

```diff
@@ -67,93 +71,50 @@ You can pass a type parameter that extends the ``Document`` interface like this:
-        :end-before: end-no-error
-
-  .. tab:: Nested Objects
-     :tabid: nested-objects
-
-     The following code snippet raises a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-error
-        :end-before: end-error
+.. _node-driver-limitations:
 
-     This is the error:
+Limitations For Driver Version {+version+}
+----------------------------------
 
-     .. code-block:: text
+You cannot specify a recursive type as a type parameter in version {+version+} of the driver.
 
-        Type 'string' is not assignable to type 'number'.
+If you specify a recursive type, the driver raises the following error:
 
-Despite the lack of type safety, we still recommend that you use dot notation to
-access nested fields in query and update documents when you use TypeScript. You
-must manually check that your nested field values have your intended type.
+.. code-block:: text
 
-.. note:: Reason To Use Dot Notation
+   error TS2615: Type of property 'r' circularly references itself in mapped type '{ [Key in keyof RecursiveType]: [Key, ...NestedPaths<RecursiveType[Key]>]; }'.
 
-   In the MongoDB Query Language, you must match a subdocument exactly
-   when specifying subdocuments in a query. Dot notation allows you to query
-   nested fields without matching subdocuments exactly.
+A recursive type is a type that references itself. You can update
+the :ref:`Pet <mongodb-node-typescript-pet-interface>` interface
+to be recursive by adding the following field:
 
-   To show this behavior, lets say you have a collection containing
-   only the following document:
-
-   .. code-block:: json
-
-      { field: { s1: "hi", s2: "bye" } }
-
-   The following query returns no results from this collection, as the value of
-   ``field`` does not exactly match ``{ s1: "hi" }``:
-
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-no-doc
-      :end-before: end-no-doc
-
-   The following queries both return your document:
+.. code-block:: typescript
+   :emphasize-lines: 2
 
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-doc
-      :end-before: end-doc
+   interface Pet {
+      pet: Pet | null;
+      name: string;
+      age: number;
+      cute: true;
+   }
 
-   The syntax of the query that does not use dot notation is cumbersome and hard
-   to understand, and may not be worth the type safety obtained from
-   avoiding dot notation.
+To track the fix for this limitation, see 
+`NODE-3852 <https://jira.mongodb.org/browse/NODE-3852>`__
+in JIRA issue tracker.
 
-For more information on dot notation, see :manual:`the MongoDB Manual </core/document/#dot-notation>`.
+If you must apply a recursive type to your documents, use version 4.2 of
```

Comment:
Should this be "classes" instead of "documents" to match the information presented in the earlier sections?

Reply:
I'll change to "classes".

Note: I think both work here. I think "classes" ties better with what is earlier in the page, but "documents" is more in-line with the task the reader is performing (enforcing structure for documents).